internal/job/verification/verification_test.go

```diff
@@ -0,0 +1,413 @@
+	err = verification.Perform()
+
+	// Assert
+	require.NoError(t, err)
+	ensureVerificationCompleted(t, cfg.finRepo, cfg.nlvRepo, cfg.fullBackupInput.ActionUID)
+}
+
+func TestVerification_Success_Resume(t *testing.T) {
+	// Description:
+	//   Perform verification after an incremental backup with interruption and check it succeeds.
+	//
+	// Arrange:
+	//   Set up a PVC and create a full backup followed by an incremental backup of it.
+	//   Then, emulate an interruption during the verification process after applying the first diff part.
+	//
+	// Act:
+	//   Perform verification on the backed-up data, resuming from the interruption.
+	//
+	// Assert:
+	//   - Check that verification completes successfully without errors.
+	//   - Confirm that the expected number of diff parts were applied during the verification process.
+	//   - Ensure that the verification action is marked as completed in the Fin repository.
+	//   - Verify that the instant verification image is removed after verification.
+
+	// Arrange
+	cfg := setup(t, &setupInput{})
+
+	fullBackup := backup.NewBackup(cfg.fullBackupInput)
+	err := fullBackup.Perform()
+	require.NoError(t, err)
+	incrementalBackup := backup.NewBackup(cfg.incrementalBackupInput)
+	err = incrementalBackup.Perform()
+	require.NoError(t, err)
+
+	require.Equal(t, 2, calculateMaxPartNumber(&job.BackupMetadataEntry{
+		SnapSize: cfg.incrementalSnapshot.Size,
+		PartSize: cfg.incrementalBackupInput.MaxPartSize,
+	}))
+
+	createFakeE2fsck(t, cfg.incrementalVolume)
+
+	// Emulate interruption after applying the first diff part
+	err = cfg.finRepo.StartOrRestartAction(cfg.incrementalBackupInput.ActionUID, model.Verification)
+	require.NoError(t, err)
+	err = cfg.nlvRepo.ReflinkRawImageToInstantVerifyImage()
+	require.NoError(t, err)
+	sourceSnapshotName, targetSnapshotName :=
+		job.CalcSnapshotNamesWithOffset(
+			cfg.fullSnapshot.Name,
+			cfg.incrementalSnapshot.Name,
+			0,
+			2,
+			cfg.incrementalBackupInput.MaxPartSize,
+		)
+	err = cfg.rbdRepo.ApplyDiffToRawImage(
+		cfg.nlvRepo.GetInstantVerifyImagePath(),
+		cfg.nlvRepo.GetDiffPartPath(cfg.incrementalSnapshot.ID, 0),
+		sourceSnapshotName,
+		targetSnapshotName,
+	)
+	require.NoError(t, err)
+	err = cfg.finRepo.UpdateActionPrivateData(cfg.incrementalBackupInput.ActionUID, []byte("{\"nextDiffPart\":1}"))
+	require.NoError(t, err)
+
+	// Act
+	verification := NewVerification(&input.Verification{
+		Repo:             cfg.finRepo,
+		RBDRepo:          cfg.rbdRepo,
+		NLVRepo:          cfg.nlvRepo,
+		ActionUID:        cfg.incrementalBackupInput.ActionUID,
+		TargetSnapshotID: cfg.incrementalBackupInput.TargetSnapshotID,
+		TargetPVCUID:     cfg.incrementalBackupInput.TargetPVCUID,
+	})
+	err = verification.Perform()
+
+	// Assert
+	require.NoError(t, err)
+	require.Equal(t, 2, cfg.rbdRepo.applyDiffCount)
+	ensureVerificationCompleted(t, cfg.finRepo, cfg.nlvRepo, cfg.incrementalBackupInput.ActionUID)
+}
+
+func TestVerification_Error_FsckFailure(t *testing.T) {
```

Comment:
Could you also add the case that `Perform()` fails and error is not `ErrFsckFailed`?

Reply:
Fixed in https://github.com/cybozu-go/fin/pull/129/commits/a5569788fc84dc997eaf553cc693b87b888b8e57.